#! /bin/sh /usr/share/dpatch/dpatch-run
## prefork_taint_abort.patch.dpatch by  <devin@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes an abort on startup caused by taint checks in qpsmtpd-prefork;
## DP: patch applied upstream but not in a formal release yet.

@DPATCH@

diff --git a/qpsmtpd-prefork b/qpsmtpd-prefork
index 882c752..3f23df3 100755
--- a/qpsmtpd-prefork
+++ b/qpsmtpd-prefork
@@ -10,6 +10,12 @@
 # safety guards
 use strict;
 
+BEGIN {
+    # secure shell
+    $ENV{'PATH'} = '/bin:/usr/bin';
+    delete @ENV{qw(IFS CDPATH ENV BASH_ENV)};
+}
+
 # includes
 use IO::Socket;
 use POSIX;
@@ -38,10 +44,6 @@ foreach my $sig_name ( split( /\s/, $Config{sig_name} ) )
     $sig_num{$sig_name} = $i++;
 }
 
-# secure shell
-$ENV{'PATH'} = '/bin:/usr/bin';
-delete @ENV{qw(IFS CDPATH ENV BASH_ENV)};
-
 # version
 my $VERSION = "1.0";
 
